argoverse/evaluation/stereo/eval.py

```diff
@@ -0,0 +1,348 @@
+    gt_obj_disparity = np.float32(gt_obj_disparity) / 256.0
+
+    errors = accumulator(abs_error_thresholds)
+
+    # Compute masks
+    pred_mask = pred_disparity > 0
+    gt_mask = gt_disparity > 0
+    obj_mask = gt_obj_disparity > 0
+    bg_mask = np.logical_and(gt_mask, np.logical_not(obj_mask))
+    fg_mask = np.logical_and(gt_mask, obj_mask)
+
+    # If the density of the predicted disparity is less than 100%, then interpolate to fill up the holes.
+    num_pixels_all = pred_disparity.size
+    num_pixels_all_est = np.sum(pred_mask)
+    density = num_pixels_all_est / max(num_pixels_all, 1.0)
+
+    if density < 1.0:
+        pred_disparity = interpolate_disparity(pred_disparity)
+
+    # Compute errors
+    abs_err = np.abs(pred_disparity - gt_disparity)
+    rel_err = abs_err / np.maximum(gt_disparity, 1)
+
+    errors["num_pixels_bg"] = np.sum(bg_mask)
+    errors["num_pixels_fg"] = np.sum(fg_mask)
+    errors["num_pixels_bg_est"] = np.sum(bg_mask & pred_mask)
+    errors["num_pixels_fg_est"] = np.sum(fg_mask & pred_mask)
+
+    for abs_error_threshold, rel_error_threshold in zip(abs_error_thresholds, rel_error_thresholds):
+        bad_pixels = (abs_err > abs_error_threshold) & (rel_err > rel_error_threshold)
+
+        errors[f"num_errors_bg:{str(abs_error_threshold)}"] = np.sum(bg_mask & bad_pixels)
+        errors[f"num_errors_fg:{str(abs_error_threshold)}"] = np.sum(fg_mask & bad_pixels)
+        errors[f"num_errors_bg_est:{str(abs_error_threshold)}"] = np.sum(bg_mask & pred_mask & bad_pixels)
+        errors[f"num_errors_fg_est:{str(abs_error_threshold)}"] = np.sum(fg_mask & pred_mask & bad_pixels)
+
+    if save_disparity_error_image:
+        err = np.minimum(abs_err / abs_error_thresholds[0], rel_err / rel_error_thresholds[0])
+        disparity_error_image = np.zeros((*pred_disparity.shape, 3), dtype=np.uint8)
+
+        for threshold, color in log_colormap:
+            disparity_error_image[np.logical_and(err >= threshold[0], err < threshold[1])] = color
+
+        disparity_error_image[gt_disparity == 0] *= 0
+
+        disparity_error_image = np.uint8(
+            cv2.dilate(disparity_error_image, kernel=np.ones((2, 2), np.uint8), iterations=3)
+        )
+
+        # Get colormap image
+        fig = plt.figure(figsize=(24, 2))
+        ax = fig.add_subplot(111)
+        x = np.linspace(1, 11, 10)
+        y = np.linspace(0, 0, 10)
+        sizes = np.logspace(8, 12, num=10, base=2.0)
+        scalars = np.array(log_colormap, dtype=object)[:, 0] * abs_error_thresholds[0]
+        scalars = [f"[{scalar[0]:.2f}, {scalar[1]:.2f}]" for scalar in scalars]
+        colors = np.array(log_colormap, dtype=object)[:, 1]
+        colors = [color / 255.0 for color in colors]
+        plt.scatter(x, y, c=colors, s=3000)
+        plt.xlabel("Disparity range errors in pixels", fontsize=24)
+        plt.xticks(x, scalars, fontsize=20)
+        plt.yticks([])
+        ax.spines["top"].set_visible(False)
```

Comment:
inline comments might make this a bit clearer -- I haven't used the `spines` attribute before

Reply:
Sure, made some changes.